node/src/daser.rs

```diff
@@ -0,0 +1,294 @@
+//! Component responsible for data availability sampling of the already synchronized block
+//! headers announced in the Celestia network.
+//!
+//! When a new header is insert in the [`Store`], [`Daser`] gets informed, then it fetches
+//! random [`Sample`]s of the block via Shwap protocol and verifies them. If all samples
+//! get verified successfuly, then block is marked as accepted.
```

Comment:
```suggestion
//! When a new header is inserted into the [`Store`], [`Daser`] gets notified. It then fetches
//! random [`Sample`]s of the block via Shwap protocol and verifies them. If all the samples
//! get verified successfuly, then block is marked as accepted.
```


Also maybe we should add something like "Otherwise, if `Daser` doesn't receive valid samples before `GET_SAMPLE_TIMEOUT`, block is marked as not accepted and dasing continues". 

I think this is implied by the first part, but you have to know how the code works to know what "get verified successfuly" means.

Reply:
Isn't this an implementation detail because of how Shwap works?

Reply:
yeah, I'd definitely drop the reference to the exact constant name. 👍 

I'm torn on the timeout, because I think it might be considered a shwap implementation detail, but it's also pretty important detail to get wrong when trying to understand DASer. 

Maybe something vaguer? "If daser is unable to get samples before shwap timeout, sample is considered unavailable, block mared as not accepted and dasing continues". Since interpretation and consequences of bitswap timeout are within DASer purview.